core/src/main/scala/filodb.core/memstore/TimeSeriesShard.scala

```diff
@@ -731,6 +731,10 @@ class TimeSeriesShard(val dataset: Dataset,
     }.flatMap {
       case Success           => blockHolder.markUsedBlocksReclaimable()
         commitCheckpoint(dataset.ref, shardNum, flushGroup)
+      // Even if data is dropped we still mark blocks as reclaimable and flush checkpoints.  We prioritize the
+      // availability of the latest metrics/data for alerting.
+      case DataDropped       => blockHolder.markUsedBlocksReclaimable()
+        commitCheckpoint(dataset.ref, shardNum, flushGroup)
```

Comment:
It is too early to make this move. We still do not know that the "Could not obtain block" error is because each and every block in the system becomes non-reclaimable. I suspect there is some other bug lurking around.

This move will result in premature data loss that is avoidable. 

Even if we decide to move forward with such an approach it should be done only after multiple flush intervals have passed without success - not immediately after one round.

Reply:
That move is independent of the bug.  But it can be separated out to a different PR.